src/main/java/ui/components/pickers/PickerMilestone.java

```diff
@@ -128,15 +130,19 @@ public int compareTo(PickerMilestone milestone) {
             return this.isOpen() ? -1 : 1;
         }
 
-        if (this.getDueDate().equals(milestone.getDueDate())) return 0;
-
         // milestones with due dates are smaller
         if (!this.getDueDate().isPresent()) return 1;
         if (!milestone.getDueDate().isPresent()) return -1;
 
         // milestones with earlier due dates are smaller
-        return this.getDueDate().get()
-                .isBefore(milestone.getDueDate().get()) ? -1 : 1;
+        LocalDate thisDueDate = this.getDueDate().get();
+        LocalDate otherDueDate = milestone.getDueDate().get();
+        if (!thisDueDate.isEqual(otherDueDate)) {
+            return thisDueDate.isBefore(otherDueDate) ? -1 : 1;
+        }
+
+        // milestones ordered according to their titles if due dates are the same
+        return this.getTitle().compareTo(milestone.getTitle());
```

Comment:
Removing equality at the top might introduce a bug, e.g. what if both milestones have no due date - currently, the other milestone will always be treated as smaller. Is this an intended behaviour?

In my opinion, for consistency, if both milestones have same due date or both have no due date, compare by titles.